bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/CleanupLedgerManager.java

```diff
@@ -111,7 +114,12 @@ private GenericCallback removeCallback(GenericCallback callback) {
 
     private void recordPromise(CompletableFuture<?> promise) {
         futures.add(promise);
-        promise.thenRun(() -> futures.remove(promise));
+        promise.whenComplete((result, exception) -> {
+            futures.remove(promise);
+            if (exception != null) {
+                LOG.error("Failed on operating ledger metadata: {}", BKException.getExceptionCode(exception));
```

Comment:
I'm not sure if we need to log the error message.

Reply:
LGTM